src/Neo/Network/P2P/Payloads/Transaction.cs

```diff
@@ -426,25 +426,23 @@ public virtual VerifyResult VerifyStateIndependent(ProtocolSettings settings)
             UInt160[] hashes = GetScriptHashesForVerifying(null);
             for (int i = 0; i < hashes.Length; i++)
             {
-                if (IsSignatureContract(witnesses[i].VerificationScript.Span))
+                if (IsSignatureContract(witnesses[i].VerificationScript.Span) && IsSingleSignatureInvocationScript(witnesses[i].InvocationScript, out ReadOnlyMemory<byte> signature))
```

Comment:
```suggestion
                if (IsSignatureContract(witnesses[i].VerificationScript.Span) && IsSingleSignatureInvocationScript(witnesses[i].InvocationScript, out var signature))
```

Reply:
Fixed.

Reply:
Also, I appreciate language-related reviewcomments or changes, because I haven't write in C# for a long time.